### PR TITLE
Sgc 2475

### PR DESCRIPTION
**Pull request by rickardabilia**
moving up the locked check and using it on line 61 is what was needed

### CODE REVIEW

File: flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/ScheduledNotificationReceiver.java [hunk 1]
```diff
@@ -50,26 +50,26 @@ public void onReceive(final Context context, Intent intent) {
...
-      boolean repeat = intent.getBooleanExtra("repeat", false);
-      if (!repeat) {
+      if (!intent.getBooleanExtra("repeat", false)) {
         FlutterLocalNotificationsPlugin.removeNotificationFromCache(context, notificationId);
       }
     } else {
-      Gson gson = FlutterLocalNotificationsPlugin.buildGson();
-      Type type = new TypeToken<NotificationDetails>() {}.getType();
+      final Gson gson = FlutterLocalNotificationsPlugin.buildGson();
+      final Type type = new TypeToken<NotificationDetails>() {}.getType();
```
Comment (bornold) [inline flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/ScheduledNotificationReceiver.java]: While I agree with all these changes, we try to keep the changes from our branch the https://github.com/MaikuB/flutter_local_notifications/tree/master to a minimum.
Because we constantly merge the main projects changes into our, it is much easier if they are as similar as possible

So could you revert the changes that is not strictly necessary?

File: flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/ScheduledNotificationReceiver.java [hunk 2]
```diff
@@ -50,26 +50,26 @@ public void onReceive(final Context context, Intent intent) {
...
         BackgroundAlarm.start(context, notificationDetails);
       }
       FlutterLocalNotificationsPlugin.scheduleNextNotification(context, notificationDetails);
 
-      boolean locked = FlutterLocalNotificationsPlugin.isKeyguardLocked(context);
-      boolean firstAlarm =
+      final boolean firstAlarm =
           LocalDateTime.parse(notificationDetails.scheduledDateTime).getSecond() == 0;
-      boolean hasStartActivity = notificationDetails.startActivityClassName != null;
+      final boolean hasStartActivity = notificationDetails.startActivityClassName != null;
```
Comment (bornold) [inline flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/ScheduledNotificationReceiver.java]: These changes are fine since this code does not exist in https://github.com/MaikuB/flutter_local_notifications/tree/master

File: flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/ScheduledNotificationReceiver.java [hunk 3]
```diff
@@ -21,18 +21,18 @@
...
   public void onReceive(final Context context, Intent intent) {
-    String notificationDetailsJson =
+    final String notificationDetailsJson =
         intent.getStringExtra(FlutterLocalNotificationsPlugin.NOTIFICATION_DETAILS);
     if (StringUtils.isNullOrEmpty(notificationDetailsJson)) {
       // This logic is needed for apps that used the plugin prior to 0.3.4
 
       Notification notification;
-      int notificationId = intent.getIntExtra("notification_id", 0);
+      final int notificationId = intent.getIntExtra("notification_id", 0);
```
Comment (bornold) [inline flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/ScheduledNotificationReceiver.java]: Revert this as well please!

**Review by bornold — CHANGES_REQUESTED** [review]
Nice!
But keep changes to minimum!

**Review by bornold — COMMENTED** [review]
Hmm, realized that doing a change request then go on vacation is a bad idea, so I am testing if commenting would remove then need for approve after request change

**Review by bornold — APPROVED** [review]
Nope, but then I'll just approve and make @LindyGPT review

File: flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/ScheduledNotificationReceiver.java (hunk 1 above)
Comment (rickardabilia) [reply flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/ScheduledNotificationReceiver.java]: Yes, that makes sense.

**Comment by LindyGPT** [comment]
Looks good to me!
